### DISCUSSION TITLE
How to set a specific font for verbatim format?

### DISCUSSION THREAD

**Discussion by Mike221223** [zim-desktop-wiki/zim-desktop-wiki#2309] [Q&A] [answered]
Hi.

(I use Zim 0.75.1 on Linux Mint 20.3)

I've run into an issue setting FreeMono (or any other specific font) as the font for verbatim format.

Here's what the my `style.conf` looks like.

`[Tag code]`
`foreground=#c8eeff`
`family=monospace`
`font=FreeMono`

I don't know if I set it correctly; I simply copied the line `font=FreeSans 13` from under `[TextView]` (which apparently depends on the font chosen in Zim GUI preferences) and pasted it under `[Tag code]`.

Text formatted in verbatim is now rendered in FreeMono, but if I apply italics or bold to it, it's still rendered as _just_ verbatim. The syntax is applied and saved, i.e it's in the source txt file, and it's rendered accordingly in the exported html files, but in Zim it's invisible. Strike and highlight work fine.

I tried replacing `monospace` with `FreeMono`, `FreeFont`, `GNU FreeFont`, removing the `family` line; none of it worked.

Can anybody tell me what I did wrong?

**Comment by hsandt** [comment, marked as answer]
I tried with Fira Code font on my side, and it worked. Case and spaces are ignored so these work:

```
family=firacode
family=fira code
family=FiraCode
family=Fira Code
family=Fi RaC od E
```

Make sure to also set it for Tag pre in case you're working with multi-line verbatim. Note that not all group of verbatim lines are multi-line verbatim. You can identify them this way:
- multi-line verbatim `pre` blocks are indented
- if you add a new line in the middle or after the last one, it will preserve indent and verbatim format

Multi-line "pre" verbatim is only enabled if you select full lines with ending newline (notice how the highlight extends to the right edge of the page), or if you manually entered surrounding `'''` (see https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1583). It may be what you did, in this case try again with the `[Tag pre]`. Remember to restart Zim.

**Reply by Mike221223** [reply]
Setting the same family for both Tag code and Tag pre worked. All formats now render according to the syntax. Thank you so much!
